### DISCUSSION TITLE
Sanitizing input

### DISCUSSION THREAD

**Discussion by nanasemanase** [pocketbase/pocketbase#3375] [Q&A] [answered]
I have a field that allows users to input HTML. I need to sanitize this HTML to prevent any potentially harmful content. One approach I'm considering is to create a quick Go-Chi endpoint that accepts the HTML input, sanitizes it, and then makes an API request to PocketBase. I am wondering if there is any other way to go about doing this.

**Comment by ganigeorgiev** [comment, marked as answer]
You can attach to the existing [`OnRecordBeforeCreateRequest`](https://pocketbase.io/docs/go-event-hooks/#onrecordbeforecreaterequest) and [`OnRecordBeforeUpdateRequest`](https://pocketbase.io/docs/go-event-hooks/#onrecordbeforeupdaterequest) hooks and call something like:
```go
app.OnRecordBeforeCreateRequest("yourCollection").Add(func(e *core.RecordCreateEvent) error {
	e.Record.Set("yourField", yourSanitizeFunc(e.Record.GetString("yourField")))

	return nil
})
```
The `yourSanitizeFunc` in the above could be [microcosm-cc/bluemonday](https://github.com/microcosm-cc/bluemonday).

---

Depending on your use case and what exactly you want to sanitize, an easier alternative could be to perform the sanitization on the client-side before rendering with something like [DOMPurify](https://github.com/cure53/DOMPurify) (_in the Admin UI you'll still see the original HTML if the `editor` field is used but it should be safe because of the default CSP policy and the XSS sanitization performed by TinyMCE_).
